src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/PopupRenderer.ios.cs

```diff
@@ -0,0 +1,288 @@
+						originY = (UIScreen.MainScreen.Bounds.Height / 2) - (PreferredContentSize.Height / 2);
+						break;
+				}
+
+				switch (Element.HorizontalOptions.Alignment)
+				{
+					case LayoutAlignment.End:
+						originX = UIScreen.MainScreen.Bounds.Width;
+						break;
+					case LayoutAlignment.Center:
+						originX = UIScreen.MainScreen.Bounds.Width / 2;
+						break;
+				}
+
+				PopoverPresentationController.SourceRect = new CGRect(originX, originY, 0, 0);
+			}
+			else
+			{
+				var view = Platform.GetRenderer(Element.Anchor).NativeView;
+				PopoverPresentationController.SourceView = view;
+				PopoverPresentationController.SourceRect = view.Bounds;
+			}
+		}
+
+		void SetBackgroundColor()
+		{
+			Control.NativeView.BackgroundColor = Element.Color.ToUIColor();
+		}
+
+		void SetView()
+		{
+			base.View.AddSubview(Control.ViewController.View);
+			base.View.Bounds = new CGRect(0, 0, PreferredContentSize.Width, PreferredContentSize.Height);
+			AddChildViewController(Control.ViewController);
+		}
+
+		void SetPresentationController()
+		{
+			((UIPopoverPresentationController)PresentationController).SourceView = ViewController.View;
+
+			// Setting PermittedArrowDirector to 0 breaks the Popover layout. It would be nice if there is no anchor to remove the arrow.
+			((UIPopoverPresentationController)PresentationController).PermittedArrowDirections = UIPopoverArrowDirection.Up;
+			((UIPopoverPresentationController)PresentationController).Delegate = new PopoverDelegate(this);
+		}
+
+		void AddToCurrentPageViewController()
+		{
+			ViewController.PresentViewController(this, true, () => Element.OnOpened());
+		}
+
+		async void OnDismissed(object sender, PopupDismissedEventArgs e)
+		{
+			if (!FormsToolkit.IsiOS9OrNewer)
+				await ViewController.DismissViewControllerAsync(true);
+			else
+				ViewController.DismissViewController(true, null);
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (isDisposed)
+				return;
+
+			isDisposed = true;
+			if (disposing)
+			{
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Platform.GetRenderer(Element) == this)
+					{
+						// NOTE - AH 9/12/2020
+						// This used to use the internal property
+						// 'Xamarin.Forms.Platform.Android.Platform.RendererProperty'
+						// That property is marked internal so the closest thing we can do
+						// is duplicate the implementation here.
+						//
+						// I don't think this is really needed for the control, but I am
+						// leaving it in so it can be reviewed.
+						var rendererProperty = BindableProperty.CreateAttached("Renderer", typeof(IVisualElementRenderer), typeof(Platform), default(IVisualElementRenderer), propertyChanged: (bindable, oldvalue, newvalue) =>
+						{
+							var view = bindable as VisualElement;
+							if (view != null)
+								view.IsPlatformEnabled = newvalue != null;
+						});
+
+						Element.ClearValue(rendererProperty);
```

Comment:
If we need to care about the clean-up, this should work, I believe...
```suggestion
						Element.ClearValue("Renderer");
```

Reply:
Does this mean we can remove the special `rendererProperty`?

Reply:
Yes, I think so.